src/main/java/org/folio/rest/impl/TransactionsApi.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.rest.impl;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.rest.util.HelperUtils.getEndpoint;
+
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.helper.AbstractHelper;
+import org.folio.rest.helper.TransactionsHelper;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.resource.Finance;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+
+public class TransactionsApi implements Finance {
+
+  private static final String TRANSACTIONS_LOCATION_PREFIX = getEndpoint(Finance.class) + "/%s";
+
+  @Validate
+  @Override
+  public void postFinanceAllocations(String lang, Transaction allocation, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    TransactionsHelper helper = new TransactionsHelper(okapiHeaders, vertxContext, lang);
+    helper.createTransaction(allocation)
```

Comment:
I believe it makes sense to add validation check like `transactionType == "Allocation"` before creating transaction. And the same checks for encumbrances and transfers.